src/NSwag.CodeGeneration.CSharp.Tests/ControllerGenerationDefaultParameter.cs

```diff
@@ -0,0 +1,113 @@
+                                Name = "bar",
+                                IsRequired = false,
+                                Kind = SwaggerParameterKind.Query,
+                                Type = JsonObjectType.Integer
+                            },
+                            new SwaggerParameter {
+                                Name = "doubledef",
+                                IsRequired = false,
+                                Default = 0.6822871999174,
+                                Kind = SwaggerParameterKind.Query,
+                                Type = JsonObjectType.Number
+                            },
+                            new SwaggerParameter {
+                                Name = "decdef",
+                                IsRequired = false,
+                                Default = 79228162514264337593543950335M,
+                                Kind = SwaggerParameterKind.Query,
+                                Type = JsonObjectType.Number,
+                                Format = JsonFormatStrings.Decimal
+                            },
+                            new SwaggerParameter {
+                                Name = "abc",
+                                IsRequired = true,
+                                Default = 84,
+                                Kind = SwaggerParameterKind.Query,
+                                Type = JsonObjectType.Integer
+                            },
+                            new SwaggerParameter {
+                                Name = "strdef",
+                                Default = @"default""string""",
+                                IsRequired = false,
+                                Kind = SwaggerParameterKind.Query,
+                                Type = JsonObjectType.String
+                            }
+                        }
+                    }
+                }
+            };
+
+            //// Act
+            var generator =
+                new SwaggerToCSharpControllerGenerator( document,
+                    new SwaggerToCSharpControllerGeneratorSettings { GenerateOptionalParameters = true,  AspNetNamespace = ""} );
+            var code = generator.GenerateFile();
+
+            //// Assert
+            Assert.IsTrue( code.Contains( "_implementation.BarAsync(abc, booldef ?? true, intdef ?? 42, doubledef ?? 0.6822871999174D, decdef ?? 79228162514264337593543950335M, strdef ?? @\"default\"\"string\"\"\", bar)" ) );
+
+            Assert.IsTrue( code.Contains( "BarAsync(int abc, bool booldef, int intdef, double doubledef, decimal decdef, string strdef, int? bar = null);" ) );
+
+            var trimmedCode = RemoveExternalReferences( code );
+
+            Console.WriteLine( trimmedCode );
+
+            CompilerParameters parameters = new CompilerParameters { GenerateInMemory = true };
+
+            var result = new CSharpCodeProvider().CompileAssemblyFromSource( parameters, trimmedCode );
```

Comment:
I didn't see this elsewhere in the controller generation code, but i think its fairly useful to verify that the generated code compiles.